src/Bicep.Core/TypeSystem/ObjectType.cs

```diff
@@ -11,13 +11,14 @@ public class ObjectType : TypeSymbol
     {
         public ObjectType(string name) : base(name)
         {
+            AdditionalProperties = LanguageConstants.Any;
         }
 
         public override TypeKind TypeKind => TypeKind.Primitive;
 
         public virtual ImmutableDictionary<string, TypeProperty> Properties => ImmutableDictionary<string, TypeProperty>.Empty;
 
-        public virtual TypeSymbol? AdditionalPropertiesType => LanguageConstants.Any;
+        public virtual ITypeReference? AdditionalProperties { get; }
```

Comment:
>AdditionalProperties [](start = 39, length = 20)

This represents the type of all additional properties. Why drop "Type" from the property name?

Reply:
Just cosmetic - otherwise you end up with `myObject.AdditionalPropertiesType.Type` instead of `myObject.AdditionalProperties.Type` to access the type.

Reply:
Hmm yeah good pt.

---
In reply to: [492362524](https://github.com/Azure/bicep/pull/536#discussion_r492362524) [](ancestors = 492362524)